src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -2129,21 +2131,29 @@ LFORTRAN_API int32_t _lpython_bit_length8(int64_t num)
 //repeat str for n time
 LFORTRAN_API void _lfortran_strrepeat(char** s, int32_t n, char** dest)
 {
-    int cntr = 0;
     char trmn = '\0';
     int s_len = strlen(*s);
     int trmn_size = sizeof(trmn);
     int f_len = s_len*n;
     if (f_len < 0)
         f_len = 0;
     char* dest_char = (char*)malloc(f_len+trmn_size);
-    for (int i = 0; i < n; i++) {
-        for (int j = 0; j < s_len; j++) {
-            dest_char[cntr] = (*s)[j];
-            cntr++;
+
+    if (s_len == 1) {
+        memset(dest_char, *(*s), n);
+    } else {
```

Comment:
I wonder why we need to handle the case for `s_len == 1` separately. The else portion seems general, so I think that should be enough to handle the case of `s_len == 1`.

Reply:
Does the else portion fail when `s_len == 1`?

Reply:
It does not fail when `s_len == `, but `memset` is very optimized for this usecase. Setting a large block  of data to a single value is what `memset` excels in, and it is optimized for every architecture. While testing, these are the compared times:
```
100000000
Binary exponentiation: 0.086963
memset: 0.006341
```
```
400000000
Binary exponentiation: 0.211737
memset: 0.018612
```
This is after compiling with `-O3`